src/lm.jl

```diff
@@ -156,7 +156,30 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(3,1) * deviance(mm)/dof_residual(mm))'
+
+    function _local(x::StridedVector)
+        Ac_ldiv_B!(R,x)
+        residvar * (x⋅x)
+    end
+
+    interval = similar(prediction)
+    for i in 1:length(interval)
+        interval[i] = _local(newx[i,:])[1]
+    end
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval)
```

Comment:
I'll try to do something like what you suggested - the reason I didn't do it just as you wrote it is that I didn't want to calculate `tvalue`, `residvar`, `prediction` and `R` repeatedly inside a loop (and the suggestion was to pass the inner version to  a loop, right)? So I defined a local function inside the scope that had access to these variables. I can also pass them explicitly to a `predict(x::StridedVector,...)` (I am doing that now) but I don't see how a user would use that?